src/final1/subtests/ExportCommandTest.java

```diff
@@ -0,0 +1,208 @@
+			optionalResults.add(leadingSpaces(line));
+		}
+
+		runs = new Run[] {
+				new ExportRun("export", expectedResults, optionalResults),
+				new NoOutputRun("quit")
+		};
+		sessionTest(runs, Input.getFile(input));
+	}
+
+	/**
+	 * Returns a matcher that accepts all strings that are {@code text} preceded with an arbitrary number of spaces
+	 * (including none).
+	 */
+	private static Matcher<String> leadingSpaces(final String expectedText) {
+		return new BaseMatcher<String>() {
+
+			@Override
+			public boolean matches(final Object testObject) {
+				String text = (String) testObject;
+				// it's unclear whether other whitespace (\s) is allowed as well
+				return text.matches(" *" + Pattern.quote(expectedText));
+			}
+
+			@Override
+			public void describeTo(Description description) {
+				description.appendText(" *" + expectedText);
+			}
+
+			@Override
+			public void describeMismatch(final Object item, final Description description) {
+				description.appendText("a string '" + expectedText + "' preceded with an arbitrary number of spaces");
+			}
+		};
+	}
+
+	class ExportRun implements Run {
+		private String command;
+		private List<Matcher<String>> expectedResults;
+		private Matcher<String> expectedMatcher;
+		private List<Matcher<String>> optionalResults;
+		private Matcher<String> optionalMatcher;
+
+		/**
+		 * Creates a test run that merges all call to {@code Terminal.printLine}. The run succeeds if the merged output
+		 * matches {@code expectedResults} line by line.
+		 * 
+		 * @param command
+		 *            The command to run
+		 * @param expectedResults
+		 *            The matchers describing the desired output
+		 */
+		public ExportRun(String command, List<Matcher<String>> expectedResults, List<Matcher<String>> optionalResults) {
+			this.command = command;
+			this.expectedResults = expectedResults;
+			this.optionalResults = optionalResults;
+			this.expectedMatcher = KitMatchers.anyOfRemaining(expectedResults);
+			this.optionalMatcher = KitMatchers.anyOfRemaining(optionalResults);
+		}
+
+		@Override
+		public String getCommand() {
+			return command;
+		}
+
+		@Override
+		public void check(String[] testedClassOutput, String errorMessage) {
+			StringBuilder mergedOutputBuilder = new StringBuilder();
+			for (String output : testedClassOutput) {
+				mergedOutputBuilder.append(output.replace("\r", ""));
+				mergedOutputBuilder.append("\n");
+			}
+			String[] outputLines = mergedOutputBuilder.toString().split("\n");
+
+			assertThat("First error at line 0: ", outputLines[0], is("digraph {"));
+			for (int i = 1; i < outputLines.length - 1; i++) {
+				String appendix = "\n First error at line " + i + ":";
+				assertThat(errorMessage + appendix, outputLines[i], anyOf(expectedMatcher, optionalMatcher));
+			}
+			assertThat("First error at last line: ", outputLines[outputLines.length - 1], is("}"));
+			if (!expectedResults.isEmpty()) {
+				fail("Did not contain these obligatory lines: " + expectedResults);
```

Comment:
This fail should at least include `errorMessage`, as now the only information the user gets is "Did not contain these obligatory lines:", but not what the input or his output was.